docs/release.md

```diff
@@ -0,0 +1,56 @@
+## vaultclient-java release
+
+### Introduction
+
+The artifacts of vaultclient-java project are stored in Nexus with the following metadata
+```
+Group ID : com.scality
+artifact ID: vaultclient-java
+```
+We separate dev packages and released packages by having ``-SNAPSHOT`` suffix or not
+in version, every git push operation will publish dev packages to [Nexus Snapshot Repositories](https://oss.sonatype.org/#view-repositories;snapshots~browsestorage),
+dev snapshot packages can be overwritten all the time and 
+can be directly used for testing in other projects.
+
+But when we want to release a package, we will need to perform the following steps:
+
+### Release Process
+
+1. Check the project version in [build.gradle](../build.gradle#L19) and [vaultclient/build.gradle](../vaultclient/build.gradle#L17),
+if the version specified is not the one you want to release, then make a PR to update it.
+
+
+2. Run release workflow in the [github actions release page](https://github.com/scality/vaultclient-java/actions/workflows/release.yml).
+
+
+3. Once the release workflow is complete successfully, proceed to Nexus Repository Manager to manually release the project.
```

Comment:
We are doing this as currently OSIS and vaultlcient-java share the same namespace and to automate closing and releasesing all previous versions should be closed and released.
# worth mentioning this as a comment with potential future improvement mentioned in the comment.